clang/test/CIR/CodeGen/builtins-memory.c

```diff
@@ -0,0 +1,56 @@
+// RUN: %clang_cc1 -triple x86_64-unknown-linux-gnu -fclangir -emit-cir %s -o %t.cir
+// RUN: FileCheck %s --check-prefix=CIR --input-file=%t.cir
+
+typedef __SIZE_TYPE__ size_t;
+void test_memcpy_chk(void *dest, const void *src, size_t n) {
+  // CIR-LABEL: cir.func @test_memcpy_chk
+  // CIR:         %[[#DEST:]] = cir.alloca {{.*}} ["dest", init]
+  // CIR:         %[[#SRC:]] = cir.alloca {{.*}} ["src", init]
+  // CIR:         %[[#N:]] = cir.alloca {{.*}} ["n", init]
```

Comment:
should we check the final LLVM output here?

Reply:
Good question. My reasoning for not doing so was that the functionality I'm adding in this diff is for `__builtin___memcpy_chk` to lower to either the `memcpy` intrinsic or the `__memcpy_chck` libcall depending on the count and size parameters, so that's all I'm testing here. Lowering `cir.libc.memcpy` to the right LLVM intrinsic or `__memcpy_chk` to an appropriate function call is done and tested elsewhere, so I thought it'd be redundant to add that testing here too. I'm open to it though depending on what people think.

Reply:
Because it's already tested elsewhere, this is fine, thanks for clarifying